network/p2p/router.go

```diff
@@ -220,6 +224,10 @@ func (r *Router) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID ui
 	return nil
 }
 
+// AppRequestFailed routes an AppRequestFailed message to the callback
+// corresponding to requestID.
+//
+// A fatal error has occurred if no corresponding requestID can be found.
```

Comment:
Maybe we should be more concrete and say that `ErrUnrequestedResponse` should be considered fatal?